src/ra_server_proc.erl

```diff
@@ -161,7 +161,7 @@ cast_command(ServerId, Priority, Cmd) ->
 -spec query(server_loc(), query_fun(),
             local | consistent | leader, timeout()) ->
     ra_server_proc:ra_leader_call_ret(term())
-    | {ok, Reply :: term(), ra_server_id() | not_known}.
+    | {ok, Reply :: term(), not_known}.
```

Comment:
To make this clearer, `ra_server_proc:ra_leader_call_ret/1` covers `{ok, Result, Leader::ra_server_id()}` already, so this definition was redundant.